packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -669,8 +669,7 @@ TargetPlatform? getTargetPlatformForName(String platform) {
     case 'web-javascript':
       return TargetPlatform.web_javascript;
   }
-  assert(platform != null);
-  return null;
+  throw Exception('Unsupported platform name "$platform"');
```

Comment:
@christopherfujino I made this throw instead of allowing a null platform.  Do you foresee any issues with that?
This is why I had to change `fuchsia` to `fuchsia-arm64` in the test since `fuchsia` is not a valid platform name.

Reply:
I thought this would be a no-op change, but now I'm wondering why that test didn't hit the assert before...

Reply:
I know, it makes me think the tests are running without asserts...